buildingmotif/label_parsing/SerializedParserMetrics.py

```diff
@@ -0,0 +1,209 @@
+import importlib.util
+import os
+import re
+
+from buildingmotif.label_parsing.build_parser import (
+    generate_parsers_for_clusters,
+    generate_parsers_for_points,
+)
+from buildingmotif.label_parsing.combinators import (
+    COMMON_EQUIP_ABBREVIATIONS_BRICK,
+    COMMON_POINT_ABBREVIATIONS,
+)
+from buildingmotif.label_parsing.tools import abbreviationsTool, codeLinter
+
+
+class SerializedParserMetrics:
```

Comment:
Let me know if you had something else in mind, but I'm wondering why this class is building the parser.  I think this metrics class is better suited as an *output* of another method which builds the parsers.

Reply:
I am not sure if I am misinterpreting, but the ```generate_parsers_for_points()``` and ```generate_parsers_for_clusters()``` methods in ```__init__``` are building the parsers, while the class just populates the relevant instance variables with the appropriate information.

Reply:
Yes, but these are called from this class. I think the entrypoint should be something like a `ParserBuilder` class  which calls those generation methods, and emits the Metrics class. We are not constructing the metrics. We are constructing parsers! The metrics are a side-effect of this

Reply:
@gtfierro The ```ParserBuilder``` class now emits a ```ParserMetrics``` class which takes care of serialization and gathering parsing related metrics. The ```ParserBuilder``` class still keeps track of the clustering and distance metrics, as that is directly related to parser generation. The notebook and tests have also been updated, let me know what you think.